File: src/test/java/leetcode75/lvl1/IsSubsequenceTest.java
package leetcode75.lvl1;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class IsSubsequenceTest {

    @Test
    void isSubsequence() {
        IsSubsequence test = new IsSubsequence();

        String s = "beg";
        String t = "abcdefg";
        assertTrue(test.isSubsequence(s, t));

        s = "axc";
        t = "ahbgdc";
        assertFalse(test.isSubsequence(s, t));

        s = "";
        t = "";
        assertTrue(test.isSubsequence(s, t));

        s = "";
        t = "some";
        assertTrue(test.isSubsequence(s, t));

        s = "some";
        t = "";
        assertFalse(test.isSubsequence(s, t));
    }
}

File: src/test/kotlin/task/kotlin/leetcode75/lvl1/IsSubsequenceTest.kt
package task.kotlin.leetcode75.lvl1

import org.junit.jupiter.api.Test

import org.junit.jupiter.api.Assertions.*

class IsSubsequenceTest {

    private val testClass: IsSubsequence = IsSubsequence()

    @Test
    fun isSubsequence() {
        var s = "abc"
        var t = "ahbgdc"
        assertTrue(testClass.isSubsequence(s, t))

        s = "axc"
        t = "ahbgdc"
        assertFalse(testClass.isSubsequence(s, t))
    }
}